CMakeLists.txt

```diff
@@ -855,3 +855,7 @@ else(NOT WIN32)
     legacy_stdio_definitions.lib Shlwapi kernel32 user32 gdi32 winspool comdlg32
     advapi32 shell32 ole32 oleaut32 uuid odbc32 odbccp32)
 endif (NOT WIN32)
+
+if (NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
+  message(FATAL_ERROR "Rippled requires a 64 bit operating system.")
+endif()
```

Comment:
I like this check because it actually checking the target architecture - but if you really want to look at the host system, I have two possibilities:

(1) me might be able to use CMAKE_HOST_SYSTEM_PROCESSOR, although I guess that doesn't necessarily guarantee that the OS is 64 bit) https://cmake.org/cmake/help/v3.6/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html

(2) we can just duplicate the python logic in cmake:

```
find_package(PythonInterp 2.7 REQUIRED)
EXECUTE_PROCESS(
    COMMAND ${PYTHON_EXECUTABLE} -c "import platform; import sys; print platform.architecture()[0]; sys.exit(1) if (platform.architecture()[0] != '64bit') else sys.exit(0);"
    RESULT_VARIABLE PY_RUN_OK
    OUTPUT_VARIABLE PY_OUT_RESULT
    OUTPUT_STRIP_TRAILING_WHITESPACE
)
if(${PY_RUN_OK})
    message(FATAL_ERROR "64 bit host platform required for building")
else()
    message("host platform looks good (${PY_OUT_RESULT}).")
endif()
```

Option 2 is a bit crude and ugly, but I think it accomplishes your goal. If there is another simple command line tool to detect platform OR something in the ENV we can use, those might be simpler alternatives.